### PR TITLE
Support more WPF usecases (nested libraries user controls, basic themes)

### PR DESCRIPTION
**Pull request by timotei**
Add almost full? support for WPF by adding support for nested libraries User Control and classes usage in XAML

If there's anything to be said is that I had to make the Res.data field mutable in order for the chain of res processors to work properly.

### CODE REVIEW

**Comment by timotei** [comment]
I've added support for Generic.xaml in Themes as well :D. A more comprehensive support will come soon, but this can be merged in the meantime if wanted.

File: ILRepack.Tests/Utils/BamlDocumentMatcher.cs [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+            _context = versionProperty + ".Minor";
+            return AreEqual(actualVersion.Minor, expectedVersion.Minor);
+        }
+
+        private bool AreEqual(object expected, object actual)
+        {
+            _expectedValue = expected;
+            _actualValue = actual;
+
+            return Equals(_expectedValue, _actualValue);
```
Comment (timotei) [inline ILRepack.Tests/Utils/BamlDocumentMatcher.cs]: The initial way I did this was with the nice *dynamic* keyword, but we don't get proper error results (e.g.: which property failed).

File: ILRepack/Steps/ResourceProcessing/BamlStreamCollector.cs [hunk 2]
```diff
@@ -54,6 +79,22 @@ public void Process(EmbeddedResource embeddedResource, ResourceWriter resourceWr
             }
         }
 
+        private void PatchGenericThemesBaml(ResourceWriter resourceWriter)
+        {
+            byte[] existingGenericBaml;
```
Comment (gluck) [inline ILRepack/Steps/ResourceProcessing/BamlStreamCollector.cs]: you don't make use of this byte[], is it for future use or should it be removed ?

Comment (timotei) [inline ILRepack/Steps/ResourceProcessing/BamlStreamCollector.cs]: Yes, I finished on friday the more complex use cases where we already have existing generic.xaml in different configurations (empty, w/ merged dictionaries and w/o merged dictionaries), but the state of the code was horrible, so will push a PR for that tomorrow ^_^

**Pull request merged**